### PR TITLE
Don't fetch the catalog page with a commit timestamp less than or equal to the lower bound

### PR DESCRIPTION
**Pull request by joelverhagen**
Blocked on https://github.com/emgarten/Sleet/pull/13.

We don't need to fetch the catalog page with a commit timestamp less than the lower bound (unless perhaps the sleet catalog has different protocol than the NuGet.org protocol.

### CODE REVIEW

File: src/NuGet.CatalogReader/CatalogReader.cs [hunk 1]
```diff
@@ -304,26 +304,40 @@ public Task<IReadOnlyList<CatalogPageEntry>> GetPageEntriesAsync(CancellationTok
...
+             *
+             * Each page in the catalog index has an associated latest commit timestamp. That is, the timestamp of the
+             * last commit made to that page. In our example above, we need to fetch pages P1, P2 and P3. 2 (our lower
+             * bound) is greater than P0's commit timestamp 1 so we can eliminate that page and lower. 6 is less than
+             * or equal to P3's commit timestamp of 7 so we take up to that page and eliminate P4 and higher. In this
+             * example, both pages will include some data we do not care about. P1 includes 2 and P3 includes 7.
+             *
+             * Note that if the upper exactly matches a commit timestamp of a page, we still fetch the next page since
+             * it's theoretically possible for two commits to have the same timestamp.
+             */
```
Comment (emgarten) [inline src/NuGet.CatalogReader/CatalogReader.cs]: Great diagram and description! 🥇

**Comment by emgarten** [comment]
Everything should be the same between nuget.org and sleet. My guess is that the original code was just sloppily getting all possible pages, this change looks much better.

I've updated master to use Sleet 2.2.0 which contains your catalog page writing fix.

**Comment by joelverhagen** [comment]
@emgarten, merged latest. CI is green now.

**Comment by emgarten** [comment]
Merged, thanks for the fix! 💯

**Comment by emgarten** [comment]
This fix is available in v1.5.8 https://www.nuget.org/packages/NuGet.CatalogReader/1.5.8

**Pull request merged**